File: examples/python/sqs_receive_message.py
"""Recibe (hasta 10) mensajes de una cola SQS sin eliminarlos todavía.

Uso: python3 sqs_receive_message.py <queue-url>
"""
import sys

import boto3

sqs = boto3.client(
    "sqs",
    endpoint_url="http://localhost:4566",
    aws_access_key_id="test",
    aws_secret_access_key="test",
    region_name="us-east-1",
)


def main():
    if len(sys.argv) < 2:
        raise SystemExit("Uso: python3 sqs_receive_message.py <queue-url>")
    queue_url = sys.argv[1]

    result = sqs.receive_message(QueueUrl=queue_url, MaxNumberOfMessages=10, WaitTimeSeconds=2)
    messages = result.get("Messages", [])
    if not messages:
        print("No hay mensajes disponibles ahora mismo.")
        return

    print(f"{len(messages)} mensaje(s) recibido(s):")
    for msg in messages:
        print(f"  - {msg['Body']} (ReceiptHandle: {msg['ReceiptHandle'][:20]}...)")
    print("\nUsa sqs_delete_message.py con el ReceiptHandle completo para confirmarlos.")


if __name__ == "__main__":
    main()

File: examples/rust/sqs_receive_message.rs
// Recibe (hasta 10) mensajes de una cola SQS sin eliminarlos todavia.
// Uso: cargo run --bin sqs_receive_message -- <queue-url>

use aws_config::BehaviorVersion;
use aws_credential_types::Credentials;
use aws_sdk_sqs::{config::Region, Client, Config};

#[tokio::main]
async fn main() -> Result<(), aws_sdk_sqs::Error> {
    let args: Vec<String> = std::env::args().collect();
    if args.len() < 2 {
        eprintln!("Uso: cargo run --bin sqs_receive_message -- <queue-url>");
        std::process::exit(1);
    }
    let queue_url = &args[1];

    let config = Config::builder()
        .behavior_version(BehaviorVersion::latest())
        .endpoint_url("http://localhost:4566")
        .region(Region::new("us-east-1"))
        .credentials_provider(Credentials::new("test", "test", None, None, "floci"))
        .build();
    let client = Client::from_conf(config);

    let result = client
        .receive_message()
        .queue_url(queue_url)
        .max_number_of_messages(10)
        .wait_time_seconds(2)
        .send()
        .await?;

    let messages = result.messages();
    if messages.is_empty() {
        println!("No hay mensajes disponibles ahora mismo.");
        return Ok(());
    }
    println!("{} mensaje(s) recibido(s):", messages.len());
    for msg in messages {
        let handle = msg.receipt_handle().unwrap_or("");
        let preview = if handle.len() > 20 { &handle[..20] } else { handle };
        println!("  - {} (ReceiptHandle: {preview}...)", msg.body().unwrap_or(""));
    }
    println!("\nUsa sqs_delete_message.rs con el ReceiptHandle completo para confirmarlos.");
    Ok(())
}
